### PR TITLE
Distinguish normal mount from format and mount in SafeFormatAndMount

### PR DESCRIPTION
**Pull request by ghost**
Renamed SafeFormatAndMount.Mount to FormatAndMount this is to avoid overloading the Mount function from the regular mounter. Some volumes such as RBD were using SafeFormatAndMount.Mount both to mount the disk and the bind mount the global dir. This works fine but when adding fsck in #14200 the problem became apparent because you cannot run fsck on a directory.

@rootfs @markturansky

### CODE REVIEW

File: pkg/util/mount/mount.go [hunk 1]
```diff
@@ -56,9 +56,11 @@ type SafeFormatAndMount struct {
...
 }
 
-// Mount mounts the given disk. If the disk is not formatted and the disk is not being mounted as read only
-// it will format the disk first then mount it.
-func (mounter *SafeFormatAndMount) Mount(source string, target string, fstype string, options []string) error {
+// FormatAndMount mounts the given disk. If the disk is not formatted
+// and it is not being mounted as read-only it will format the disk
+// first then mount it. If the disk is already formatted or it is
+// being mounted as read-only it will be mounted without formatting.
+func (mounter *SafeFormatAndMount) FormatAndMount(source string, target string, fstype string, options []string) error {
```
Comment (rootfs) [inline pkg/util/mount/mount.go]: Can you put more words on usage? Distinguishing `Mount` and `SafeFormatAndMount` doesn't look quite easy to digest on the first glance.

Comment (swagiaal) [inline pkg/util/mount/mount.go]: @rootfs updated... do fc, iscsi, and rbd changes look OK to you ?

Comment (rootfs) [inline pkg/util/mount/mount.go]: @swagiaal look good to me. spent some time figuring out ident in `fc` though

File: pkg/util/mount/mount.go [hunk 2]
```diff
@@ -49,16 +49,20 @@ type MountPoint struct {
...
+// mounts it otherwise the device is formatted first then mounted.
 type SafeFormatAndMount struct {
 	Interface
 	Runner exec.Interface
 }
 
-// Mount mounts the given disk. If the disk is not formatted and the disk is not being mounted as read only
-// it will format the disk first then mount it.
-func (mounter *SafeFormatAndMount) Mount(source string, target string, fstype string, options []string) error {
+// FormatAndMount formats the given diskm if needed, and mounts the
```
Comment (markturansky) [inline pkg/util/mount/mount.go]: typo: diskm

File: pkg/util/mount/mount.go [hunk 3]
```diff
@@ -49,16 +49,20 @@ type MountPoint struct {
...
+// given disk. That is if the disk is not formatted and it is not
```
Comment (markturansky) [inline pkg/util/mount/mount.go]: redundant "given disk".

maybe more succinct as "FormatAndMount formats the given disk, if needed, before mounting."?

File: pkg/volume/iscsi/iscsi.go [hunk 4]
```diff
@@ -105,10 +105,10 @@ func (plugin *iscsiPlugin) newBuilderInternal(spec *volume.Spec, podUID types.UI
 			iqn:     iscsi.IQN,
 			lun:     lun,
 			manager: manager,
-			mounter: &mount.SafeFormatAndMount{mounter, exec.New()},
 			plugin:  plugin},
 		fsType:   iscsi.FSType,
 		readOnly: readOnly,
+		mounter:  &mount.SafeFormatAndMount{mounter, exec.New()},
```
Comment (markturansky) [inline pkg/volume/iscsi/iscsi.go]: I know I've seen `exec.New()` elsewhere in the code, so a future PR, perhaps, but it's probably better for plugins to obtain an exec interface from the plugin host.

Comment (swagiaal) [inline pkg/volume/iscsi/iscsi.go]: @markturansky Yeah.. this actually will need to be mocked when adding fsck in #14200 so I'll make it an argument to the `newBuilderInternal` and perhaps use the plugin's interface in `NewBuilder`.

**Comment by swagiaal** [comment]
@thockin PTAL

**Comment by swagiaal** [comment]
@thockin or @saad-ali could one of you guys review this ?

**Comment by saad-ali** [comment]
Yep, I'll take a look today.

File: pkg/util/mount/mount.go [hunk 5]
```diff
@@ -49,16 +49,20 @@ type MountPoint struct {
...
 	Interface
 	Runner exec.Interface
 }
 
-// Mount mounts the given disk. If the disk is not formatted and the disk is not being mounted as read only
-// it will format the disk first then mount it.
-func (mounter *SafeFormatAndMount) Mount(source string, target string, fstype string, options []string) error {
+// FormatAndMount formats the given disk, if needed, and mounts it.
+// That is if the disk is not formatted and it is not being mounted as
+// read-only it will format the it first then mount it. Otherwise, if
```
Comment (saad-ali) [inline pkg/util/mount/mount.go]: "format the it first" -> "format it first"

**Comment by saad-ali** [comment]
LGTM mod nit

**Comment by swagiaal** [comment]
@saad-ali comment addressed. PTAL.

**Comment by saad-ali** [comment]
LGTM

**Comment by swagiaal** [comment]
@k8s-bot test this

**Pull request merged**